### ISSUE TITLE
Freefilesync upgrade

### ISSUE THREAD

**Issue by adripo** [teknowledgist/Chocolatey-packages#81]
Package id: freefilesync

Current package version: 10.2
Version to update to: 12.0

**Comment by adripo** [comment]
I just saw that the package is in the discontinued folder. May I ask you what is the reason?

**Comment by teknowledgist** [comment]
TLDR: I can't

#### See  
https://community.chocolatey.org/packages/freefilesync#description   

##### and
https://community.chocolatey.org/packages/freefilesync#discussion

### Details:   
I get requests to update this package regularly.  I am still quite active in the Chocolatey Community, so the reason FreeFileSync is "discontinued" and not up-to-date is certainly not that I am no longer interested.

Let me try to explain the situation...

FreeFileSync source is open; It is available to download.  The FreeFileSync installer is not open, and the author, Zenju, has changed 
their license to prevent redistribution of the installer. (IOW, it cannot be an embedded package.)  It is my belief (without evidence) that this was done because the Chocolatey package was popular and the author gets advertisement money when users visit the site to download the installer.

 With most Chocolatey packages, a license that doesn't allow redistribution would not be a problem because there are tools to assist downloading the software.  In fact, before the license changed (and before it was turned into an embedded package), Zenju had designed the website to make it very difficult to perform a scripted download (again, I believe to force users' eyeballs onto the site).  Before the license change, the free version installer was designed to NOT be silently installed.  I spent many hours figuring out how to get around that via AutoHotKey, and it was a real challenge given that the button names were randomized, and on some screens the button order was randomized.  

It appears to me (again without real evidence) that once Zenju figured out that it was difficult to prevent silent install through technical means, the installer changed (and possibly the license again) to start identifying "headless" installs.  A big warning pops up about how it is a violation of the license to try to get around the manual install for the free version and that violators will have legal action taken against them.  Interestingly, once that changed, the download became scriptable again.  I can now easily use an update.ps1 script for AU that identifies and downloads the latest version.  I just can't legally install it.

If you can see a way to get around the legal obligations and technical obstacles, please let me know and I will be happy to continue the package.  The only way I can imagine would be to download the source and script the compilation.  I don't think that is reasonable as the compiler would need to be a dependency, and within the source code there are a number of identified issues that require downgrading a small number of libraries before compilation.  That would mean writing a parser for the exceptions 
list, and the whole thing just gets very, very messy and bloated.

Ultimately, the author does not want there to be a Chocolatey (or other package manager) package, and that is their right even if it seems childish and silly to us.

**Comment by adripo** [comment]
Thank you very much for the explanation.
I think this is also related to the 'automatic update' feature that is available with the [Donation edition](https://freefilesync.org/faq.php#donation-edition).

As you said, without the author's consent, this process cannot be easily achieved.

Another alternative would be to provide a local chocolatey package that one can manually install on its own machine and update with au.

I am closing the issue for now.
